draft-polli-ratelimit-headers.md

```diff
@@ -383,6 +378,43 @@ after the moment referenced by `RateLimit-Reset`.
 The server MAY arbitrarily alter the `RateLimit-Reset` value between subsequent requests
 eg. in case of resource saturation or to implement sliding window policies.
 
+## RateLimit-Scope {#ratelimit-scope}
+
+The scope of the RateLimit headers may be the target resource,
+its parent path or the whole Origin.
+
+The RateLimit-Scope field can be used to convey the URI or path
+associated￼to the returned RateLimit headers.
+
+The field value is
+
+~~~ abnf
+   RateLimit-Scope = URI-reference
+~~~
+
+This field MUST NOT occur multiple times;
+if a user agent receives multiple
+RateLimit-Scope fields, then it SHOULD ignore them.
```

Comment:
Not willing to open a can of worms, but this made me wonder whether there could be a reasonable use case for multiple occurrences. Maybe the case in which business logic imposes a rate limit but it is further limited by a reverse proxy that is too loaded? In such cases, _maybe_ the information can be coalesced into a single header for both limiting information as well as scoping.

Edit: ok, kind of answered below in the reference to proxies modifying the URI reference.

Reply:
The only reason for multiple ratelimit-scope would be to mimick the `RateLimit-Limit` field passing a different scope for every policy.

The other solution (eg. the `scope` parameter in RateLimit-Limit) is probably more flexible but wouldn't that be too complex to implement?

Reply:
As long as you can attach a scope to each policy the complexity will remain similar. If only one scope is provided when specifying multiple policies, the question would arise as to whether the scope applies to all of them or only one of them (first one?).

Additionally, what would be the relation between this scope header and other efforts in scoping such as the one on Retry-After? Do you think feasible to have a definition of scoping that covers both the rate limiting and retry-after cases?

If anything, having the option of attaching a scope to each policy being returned is nice-to-have.